opentracing-examples/src/test/java/io/opentracing/examples/async_propagation/Promise.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing.examples.async_propagation;
+
+import io.opentracing.ActiveSpan;
+import io.opentracing.mock.MockTracer;
+import io.opentracing.tag.Tags;
+
+import java.util.Collection;
+import java.util.LinkedList;
+
+/** @author tylerbenson */
+public class Promise<T> {
+  private final PromiseContext context;
+  private final MockTracer tracer;
+  private final ActiveSpan activeSpan;
+
+  private final Collection<Pair<SuccessCallback<T>>> successCallbacks = new LinkedList<>();
```

Comment:
`Collection` should be `List`
 `LinkedList` usually bad idea to use, `ArrayList` is faster.

Reply:
`Collection` was used because the order of the callbacks is irrelevant. `LinkedList` was used because I am only ever appending or iterating through the results, so random access is unneeded.  I realize that `ArrayList` is probably faster since we never exceed the default size, so the `LinkedList` allocation is slower in comparison.

That being said, like with the `Actor` class, this is not intended as a class you would use outside of the scope of this test.